provider/pkg/provider/provider_defaultTags_test.go

```diff
@@ -0,0 +1,46 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDefaultTags(t *testing.T) {
```

Comment:
There are some more cases that would be good to cover IMO, like not overwriting existing tags with default tags.